File: resource/java/com/template/Keji.java
package com.template;
import java.io.DataInputStream;
public class Keji implements TemplateInterface
{
	public String id;
	public String name;
	public int level;
	public String des;
	public String icon;
	public int shengwang_level;
	public int keji_level;
	public String[] goods;
	public int time;
	public String[] effect;
	public void read(DataInputStream in)throws Exception
	{
		id=in.readUTF();
		name=in.readUTF();
		level=in.readInt();
		des=in.readUTF();
		icon=in.readUTF();
		shengwang_level=in.readInt();
		keji_level=in.readInt();
		short goods_count=in.readShort();
		if(goods_count>0)
		{
			goods=new String[goods_count];
			for(int i=0;i<goods_count;i++)
			{
				goods[i]=in.readUTF();
			}
		}

		time=in.readInt();
		short effect_count=in.readShort();
		if(effect_count>0)
		{
			effect=new String[effect_count];
			for(int i=0;i<effect_count;i++)
			{
				effect[i]=in.readUTF();
			}
		}

	}
}

File: resource/cpp/com/template/Keji.cpp
#include "Keji.h"
void Keji::read(ByteBuffer* buffer)
{
		this->id=buffer->readUTF();
		this->name=buffer->readUTF();
		this->level=buffer->readInt();
		this->des=buffer->readUTF();
		this->icon=buffer->readUTF();
		this->shengwang_level=buffer->readInt();
		this->keji_level=buffer->readInt();
		short goods_count=buffer->readShort();
		if(goods_count>0)
		{
			for(int i=0;i<goods_count;i++)
			{
				std::string tmp=buffer->readUTF();
				this->goods.push_back(tmp);
			}
		}

		this->time=buffer->readInt();
		short effect_count=buffer->readShort();
		if(effect_count>0)
		{
			for(int i=0;i<effect_count;i++)
			{
				std::string tmp=buffer->readUTF();
				this->effect.push_back(tmp);
			}
		}

}
